### DISCUSSION TITLE
Access pageContext in prerender()

### DISCUSSION THREAD

**Discussion by joaocosteira** [vikejs/vike#1029] [Discussion] [answered]
Hello, first of all I must say that I already saw #722 and the [i18n](https://vite-plugin-ssr.com/i18n#pre-rendering) docs, but I haven't figured this thing out so I'm seeking some advice.
The idea is simple, I would like to be able to access the pageContext withing the prerender function, since the routing/urls depend on that particular data, pretty much my end goal is to being able ssg a bunch of pages, with the same data, but with different sizes.

What I'm doing:
```ts
import { prerender } from 'vite-plugin-ssr/prerender';
await prerender({
        viteConfig: { ...config },
          pageContextInit: {
              pageProps: {
                  content: ...,
                  sizes: [ ... ]
          }
    } 
})
```
What I wanted to be able to do:
```ts
export { prerender }
async function prerender(pageContext) {
  const { pageProps } = pageContext;
  return [...pageProps.sizes.map((size) => `/${size.x}x${size.y}`)]
}
```
And then ssg the content according to the size. I'm able to do that by using a pre saved list or by calling an endpoint like it is shown in the docs, but I would like to be able to access the pageContext within the prerender, more specifically the data passed through the vite-plugin-ssr/prerender function. Based on what I understood, the onBeforePrerender is actually called after the render function.

**Comment by brillout** [comment, marked as answer]
If I correctly understand your use case, I think you can achieve that by only using `prerender()` + `onBeforeRoute()`. In other words, you only use `prerender()` to generate your variants and their `pageContext`, instead of using `onBeforePrerender()`.

> Based on what I understood, the onBeforePrerender is actually called after the render function.

That is correct. (The name is quite confusing and [will be renamed](https://vite-plugin-ssr.com/migration/v1-design#renamed-hooks).)

> end goal is to being able ssg a bunch of pages, with the same data, but with different sizes.

Curious: why not responsive pages instead?

**Comment by joaocosteira** [comment]
Hello, sorry for the late reply but I was away.

Long story short, by analyzing the examples/i18n project I was able to do what I wanted (with the onBeforePrerender function, I can access the original context and then by updating the urlOriginal property, I can generate all the different routes. Thank you for that examples.

Answering your question, I simply need to ssg a bunch of examples of the same content with different styling/sizing, since I'm using the prerender programmatic function, the list of sizes is provided via props with the pageContextInit, instead of already knowing them in before hand (i.e. saved in a local file) or calling an additional endpoint (like we have in the movies example in the docs). That's why I needed a hack to generate them and that i18n example is perfect (by update the urlOriginal, for instance we ssg not only the about page, but the about page in german/french/etc), so I pretty much needed to do the same thing but with sizes.

Thank you.

**Reply by brillout** [reply]
👍 Btw. since you're using VPS in ways that you can't with other frameworks (I'm not sure you could even achieve what you have), sopnsoring can make a lot of sense for your company. (I'll also add your use case to VPS's CI.)
